### PR TITLE
docs: Improve natspec for LSP0

### PR DESCRIPTION
**Pull request by YamenMerhi**
# What does this PR introduce?

## 📄 Documentation

docs: Improve natspec for LSP0

### PR Checklist

- [ ] Wrote Tests
- [ ] Wrote & Generated Documentation (readme/natspec/dodoc)
- [ ] Ran `npm run lint` && `npm run lint:solidity` (solhint)
- [ ] Ran `npm run format` (prettier)
- [ ] Ran `npm run build`
- [ ] Ran `npm run test`

### CODE REVIEW

File: contracts/LSP0ERC725Account/LSP0ERC725AccountCore.sol [hunk 1]
```diff
@@ -787,14 +799,7 @@ abstract contract LSP0ERC725AccountCore is
...
      *
-     * @custom:hint This function does not forward to the extension contract the `msg.value` received by the contract that inherits `LSP17Extendable`.
-     * If you would like to forward the `msg.value` to the extension contract, you can override the code of this internal function as follow:
-     *
-     * ```solidity
-     * (bool success, bytes memory result) = extension.call{value: msg.value}(
-     *     abi.encodePacked(callData, msg.sender, msg.value)
-     * );
-     * ```
+     * @custom:hint If you would like to forward the `msg.value` to the extension contract, you should store an additiona `0x01` byte after the address of the extension under the corresponding LSP17 data key.
```
Comment (CJ42) [inline contracts/LSP0ERC725Account/LSP0ERC725AccountCore.sol:802]:
```suggestion
     * @custom:hint If you would like to forward the `msg.value` to the extension contract, you should store an additional `0x01` byte after the address of the extension under the corresponding LSP17 data key.
```

File: contracts/LSP0ERC725Account/LSP0ERC725AccountCore.sol [hunk 2]
```diff
@@ -89,11 +89,19 @@ abstract contract LSP0ERC725AccountCore is
     using Address for address;
 
     /**
+
+     *
+     * @custom:events {UniversalReceiver} event when receiving native tokens.
```
Comment (CJ42) [inline contracts/LSP0ERC725Account/LSP0ERC725AccountCore.sol:94]: Why is there this Natspec tag and comment block? It is already present below

image

File: contracts/LSP0ERC725Account/LSP0ERC725AccountCore.sol [hunk 3]
```diff
@@ -89,11 +89,19 @@ abstract contract LSP0ERC725AccountCore is
...
+     */
+
+    /**
+     * @notice Invokes the `universalReceiver` function logic internally when receiving native tokens.
```
Comment (CJ42) [inline contracts/LSP0ERC725Account/LSP0ERC725AccountCore.sol:98]: This is not a `@notice` tag, but documentation for developers. Suggesting adding this under the `@dev` tag.

File: contracts/LSP0ERC725Account/LSP0ERC725AccountCore.sol [hunk 4]
```diff
@@ -120,6 +128,8 @@ abstract contract LSP0ERC725AccountCore is
      *
      * 2. If the data sent to this function is of length less than 4 bytes (not a function selector), return.
      *
+     * Whenever the call is associated with native tokens, the function will delegate internally the handling of native tokens to the {universalReceiver} function passing `_TYPEID_LSP0_VALUE_RECEIVED` as typeId and the calldata as received data, except when the native token will be sent directly to the extension.
```
Comment (CJ42) [inline contracts/LSP0ERC725Account/LSP0ERC725AccountCore.sol:131]: Also worth mentioning when receiving less than 4 bytes of data.

Comment (CJ42) [reply contracts/LSP0ERC725Account/LSP0ERC725AccountCore.sol:131]: > except when the native token will be sent directly to the extension.

This last sentence makes it hard to understand. The only exception according to the code is when:
- you receive some native tokens
-  you receive some data that is more than 4 bytes.

In this case, the associated LSP17 extension is called.

**Review by CJ42 — CHANGES_REQUESTED** [review]

File: contracts/LSP0ERC725Account/LSP0ERC725AccountCore.sol (hunk 4 above)
Comment (YamenMerhi) [reply contracts/LSP0ERC725Account/LSP0ERC725AccountCore.sol:131]: This is generic saying all the time when native token is associated (even when data is less than 4 bytes), the universalReceiver is invoked, unless the native token is being forwarded to the extension, but happy to add a suggestion if you have one
> Whenever the call is associated with native tokens, the function will delegate internally the handling of native tokens to the {universalReceiver} function passing `_TYPEID_LSP0_VALUE_RECEIVED` as typeId and the calldata as received data, except when the native token will be sent directly to the extension.

**Review by CJ42 — APPROVED** [review]

**Pull request merged**